modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/exportimport/data/handler/VulcanBatchEnginePortletDataHandler.java

```diff
@@ -90,6 +95,50 @@ public boolean isCompany() {
 		return true;
 	}
 
+	protected Map<String, Serializable> buildParameterMap(
```

Comment:
I am not sure about making a method `protected` just for testing purposes.

In your case, I think it is possible to test the public method `doExportData` by passing different `portletDataContext` mocks and then, verify the `addBatchEngineExportTask` has been called with the expected parameters in each case.

Reply:
@ccorreagg thank you for the :eyes: . Here just a note: I think it gets a too complicated without PowerMock - which I'm not sure we are using? Private methods are called there. See for example `_getBytes(),` the `doExportData() `calls.

On the other hand, I don't personally see there added value  instead of just checking that the filter string is correctly built.  I'll add a Playwright test for end to end to complement this and see that the Object resourceimpl translates it correctly.

Good or not, I can see `protected` used a lot for these purposes in product tests.